src/runtime_src/core/pcie/driver/linux/xocl/subdev/ert_ctrl.c

```diff
@@ -556,29 +556,29 @@ static void ert_ctrl_legacy_fini(struct ert_ctrl *ec)
 	return;
 }
 
-static void ert_ctrl_unset_xgq(struct platform_device *pdev)
+static void ert_ctrl_unset_xgq(struct platform_device *pdev, void *xgq_handle)
 {
 	struct ert_ctrl *ec = platform_get_drvdata(pdev);
 	xdev_handle_t xdev = xocl_get_xdev(pdev);
-	int i = 0;
+	int xgq_id = xocl_get_xgq_id(xgq_handle);
+	struct ert_ctrl_xgq_cu  *xgq_ips = NULL;
 
-	for (i = 0; i < ec->ec_num_xgq_ips; i++) {
-		struct ert_ctrl_xgq_cu  *xgq_ips = &ec->ec_xgq_ips[i];
+	if (xgq_id == -EINVAL)
+		return;
```

Comment:
If xgq_id is -EINVAL here, it is a BUG_ON(). Not silently return.

Reply:
Sure. Updated. Thanks